### PR TITLE
Clean up examples, make it more "Rusty"

### PR DESCRIPTION
**Pull request by nyurik**

### CODE REVIEW

File: examples/connect5.rs [hunk 1]
```diff
@@ -651,10 +636,8 @@ fn eval(pos: &Pos, _ply: i32) -> i32 {
                     return 2560;
                 }
             }
-        } else {
-            if check_patterndead4(pos, atk) > 0 {
```
Comment (Amanieu) [inline examples/connect5.rs:655]: I'm not a fan of this clippy lint, I think in many cases it actually makes the code *less* readable.

Comment (nyurik) [reply examples/connect5.rs:655]: @Amanieu which clippy lint are you not a fan of?  Most of these are simple stylistic changes that are pretty standard in the Rust code... to my knowledge

Comment (Amanieu) [reply examples/connect5.rs:655]: Specifically the one that turns `else { if X {} }` into `else if X {}`. I often find that the additional nesting improves readability, especially if the `X` condition is completely unrelated to the condition in the outer `if`.

Comment (Amanieu) [reply examples/connect5.rs:655]: Could you undo the change to `else` here and above?

**Comment by nyurik** [comment]
thx, fixed

**Pull request merged**